### DISCUSSION TITLE
PR approval process

### DISCUSSION THREAD

**Discussion by xmulligan** [lfn-cnti/bestpractices#74] [General]
Idea:
Below are the minimums for getting a PR approved and merged into the CNF WG

Use cases: 1 tech lead  + 2 community members
For charter updates 2 chairs + 1 tech lead + 3 community members
For best practices 2 tech lead + 3 community members

**Comment by xmulligan** [comment]
Charter updates would need to exclude interested parties, otherwise a company could theoretically being blocked from participating which would not be in the interest of the community.

**Reply by xmulligan** [reply]
Maybe adding interested parties only needs one community member approval?

**Reply by iawells** [reply]
Charter updates: _if_ we consider tech leads and committee members (as elected officials) to represent the will of the people, _then_ I would suggest a charter update would be a quorum + majority setup.

Use cases, best practices: should they be the same or different?  It's clear that I could have two use cases that are in conflict, so potentially more of them; could I have best practices that contradict each other, or would one best practice necessarily exclude another?  If so, best practices might (as @xmulligan says) need more checks and balances.

Would there be baselines of best practices (e.g. 1 + 2 + 4 + 7 + 8 == a specific set for a specific purpose)?  This might be, for instance, that CNF devs would have a set of best practices they would apply in CI that makes their job easier but ultimately does not affect an operator's use of the product.  The reason I ask is that that might change what I just said about best practices forming a self-consistent whole, because that sort of coding/dev best practice is more often 'here are a list of good ideas, choose the ones that work for you that you find practical to implement' - more a cost/benefit thing.

**Comment by taylor** [comment]
Everything else:
minimum of 3 approvals

**Comment by xmulligan** [comment]
[Watson] There should be a role with veto power

**Reply by iawells** [reply]
> [Watson] There should be a role with veto power

We have three chairs, and I imagine we're thinking of a presidential-style veto. If you need 2/3 for something, you are setting a pretty high bar without being beholden to a single person.  If you make any of them a veto, you're really asking for 3/3.

**Comment by iawells** [comment]
I would consider a time limit for anything the chairs do - if you're waiting for 2 chairs to agree you could be waiting for quite some time.

**Comment by electrocucaracha** [comment]
Maybe another option can be add equivalents like 1 chair's vote can be replaced by 2 or 3 tech leads votes

**Comment by vukg** [comment]
Hi there,
reflecting on the  discussion in WG call today regarding voting and over representation I had following thoughts:
- each best practice shall be connected to real user story and/or use case.
- user stories and use cases are proposed by the people who care about that use case and want the points addressed.
- the best practice shall effectively solve one or more pain points from the user story / use case.
- there shall be consensus that the best practice actually solves the pain points and it shall be testable as part of conformance testing.
- there can be more best practices that solve same pain point, each with their own pros and cons

Having said that, should we also not let the reporter of the user story / use case and other community members who can sufficiently prove that they are relevant actors in such user story / use case in real life to vote on best best practices related to it? This way we are making sure that best practice is actually beneficial from end-user perspective and enabling to have more than one best practice. It sort of creates market for best practices from various parties and adoption decides which one is really the "best" and which one is not.

As we accumulate the best practices some will over time positively or negatively influence some other and it will be also one measure of how "best" the practice is.

With all of that we go away from standardization battles and voting as the measure of valuation towards decisions driven by intrinsic end-user / relevant actors to solve the pain points in meaningul way.

What do you think?

**Reply by xmulligan** [reply]
I tried to add community member approvers to the process to address this, but the hard part might be deciding who is eligible to say they are affected the issue. I really like the idea, I'm just not sure the best way to implement it
